### PR TITLE
AICORE-337: replacing common objects with SDK

### PR DESCRIPTION
**Pull request by andreinechaev**

### CODE REVIEW

File: nuxeo-ai-core/src/test/java/org/nuxeo/ai/bulk/TensorTest.java [hunk 1]
```diff
@@ -233,21 +231,20 @@ public void testBadDescriptor() {
...
-                "\t<li>run the following SQL query against your database\n" +
-                "<div style=\"\\\"border-width:\" class=\"\\\"preformatted\"><div class=\"\\\"preformattedContent\">\n" +
-                "<pre>select * from versions where versionableid='cd551cce-5322-4368-b045-e13086f250bb';\n" +
-                " id | versionableid | created | label | description | islatest | islatestmajor \n" +
-                "--------------------------------------------------------------------------------------" +
-                "------------------------------------------------\\n" +
-                " ba10d4af-9045-48bc-be46-0c293b84b408 | cd551cce-5322-4368-b045-e13086f250bb | 2016-05-26 14:38:00.815 | | | t | f\n" +
-                "</pre>\n</div></div>\n<p>=> the label value is empty</p></li>\n</ul>\n" +
-                "\n\n<p>This value is returned when calling <tt>session.getLastVersion(docRef).getLabel()</tt>.</p>";
+        String html = "<p>Step to reproduce:</p>\n"
```
Comment (jcarsique) [inline nuxeo-ai-core/src/test/java/org/nuxeo/ai/bulk/TensorTest.java:234]: Maybe easier to maintain if it was moved to a resource file.

**Comment by andreinechaev** [comment]
Still need to realign on the sprint branch

**Review by jcarsique — APPROVED** [review]

File: nuxeo-ai-core/src/test/java/org/nuxeo/ai/bulk/TensorTest.java (hunk 1 above)
Comment (andreinechaev) [reply nuxeo-ai-core/src/test/java/org/nuxeo/ai/bulk/TensorTest.java:234]: true that

File: nuxeo-ai-model/pom.xml [hunk 2]
```diff
@@ -16,6 +16,16 @@
       <groupId>org.nuxeo.ai</groupId>
       <artifactId>nuxeo-ai-core</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.nuxeo.ai</groupId>
+      <artifactId>insight-rest-client</artifactId>
+      <scope>compile</scope>
```
Comment (vpasquier) [inline nuxeo-ai-model/pom.xml:22]: as far as i remember compile is the default scope (so no need to reference it)

File: nuxeo-ai-model/src/main/java/org/nuxeo/ai/cloud/NuxeoCloudClient.java [hunk 3]
```diff
@@ -342,58 +314,64 @@ private AICorpus createCorpus(DocumentModel datasetDoc, String batch1, String ba
...
+        String datasource = null;
+        if (getClient() != null) {
+            datasource = client.getConfiguration().getDatasource();
+        }
+        return getModels(API.Model.PUBLISHED, Collections.singletonMap(LABEL_PARAM, datasource));
     }
 
-    protected JSONBlob getModels(String parameters) throws IOException {
-        if (!isReady()) {
+    protected JSONBlob getModels(API.Model endpoint, Map<String, Serializable> params) throws IOException {
```
Comment (vpasquier) [inline nuxeo-ai-model/src/main/java/org/nuxeo/ai/cloud/NuxeoCloudClient.java:363]: Is it not something to put inside of the insight SDK/client all this part?

**Review by vpasquier — APPROVED** [review]

File: nuxeo-ai-model/pom.xml (hunk 2 above)
Comment (andreinechaev) [reply nuxeo-ai-model/pom.xml]: ok, thanks

File: nuxeo-ai-model/src/main/java/org/nuxeo/ai/cloud/NuxeoCloudClient.java (hunk 3 above)
Comment (andreinechaev) [reply nuxeo-ai-model/src/main/java/org/nuxeo/ai/cloud/NuxeoCloudClient.java:363]: I would like to keep it as a helper method. I believe the sdk has clear method of calling endpoints

**Pull request merged**